feature/bgp/policybase/otg_tests/nested_policies/nested_policies_test.go

```diff
@@ -0,0 +1,779 @@
+	"github.com/openconfig/featureprofiles/internal/deviations"
+	"github.com/openconfig/featureprofiles/internal/fptest"
+	"github.com/openconfig/featureprofiles/internal/otgutils"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/gnmi"
+	"github.com/openconfig/ondatra/gnmi/oc"
+	otgtelemetry "github.com/openconfig/ondatra/gnmi/otg"
+	"github.com/openconfig/ygnmi/ygnmi"
+	"github.com/openconfig/ygot/ygot"
+)
+
+const (
+	ipv4PrefixLen     = 30
+	ipv6PrefixLen     = 126
+	v41Route          = "203.0.113.0"
+	v41TrafficStart   = "203.0.113.1"
+	v42Route          = "198.51.100.0"
+	v42TrafficStart   = "198.51.100.1"
+	v4RoutePrefix     = uint32(24)
+	v61Route          = "2001:db8:128:128::0"
+	v61TrafficStart   = "2001:db8:128:128::1"
+	v62Route          = "2001:db8:128:129::0"
+	v62TrafficStart   = "2001:db8:128:129::1"
+	v6RoutePrefix     = uint32(64)
+	dutAS             = uint32(65656)
+	ateAS             = uint32(65657)
+	bgpName           = "BGP"
+	maskLenExact      = "exact"
+	localPref         = 200
+	med               = 1000
+	v4Flow            = "flow-v4"
+	v4PrefixPolicy    = "prefix-policy-v4"
+	v4PrefixStatement = "prefix-statement-v4"
+	v4PrefixSet       = "prefix-set-v4"
+	v4LPPolicy        = "lp-policy-v4"
+	v4LPStatement     = "lp-statement-v4"
+	v4ASPPolicy       = "asp-policy-v4"
+	v4ASPStatement    = "asp-statement-v4"
+	v4MedPolicy       = "med-policy-v4"
+	v4MedStatement    = "med-statement-v4"
+	v6Flow            = "flow-v6"
+	v6PrefixPolicy    = "prefix-policy-v6"
+	v6PrefixStatement = "prefix-statement-v6"
+	v6PrefixSet       = "prefix-set-v6"
+	v6LPPolicy        = "lp-policy-v6"
+	v6LPStatement     = "lp-statement-v6"
+	v6ASPPolicy       = "asp-policy-v6"
+	v6ASPStatement    = "asp-statement-v6"
+	v6MedPolicy       = "med-policy-v6"
+	v6MedStatement    = "med-statement-v6"
+)
+
+var (
+	dutPort1 = attrs.Attributes{
+		Desc:    "dutPort1",
+		IPv4:    "192.0.2.1",
+		IPv4Len: ipv4PrefixLen,
+		IPv6:    "2001:db8::192:0:2:1",
+		IPv6Len: ipv6PrefixLen,
+	}
+
+	atePort1 = attrs.Attributes{
+		Name:    "atePort1",
+		MAC:     "02:00:01:01:01:01",
+		IPv4:    "192.0.2.2",
+		IPv4Len: ipv4PrefixLen,
+		IPv6:    "2001:db8::192:0:2:2",
+		IPv6Len: ipv6PrefixLen,
+	}
+
+	dutPort2 = attrs.Attributes{
+		Desc:    "dutPort2",
+		IPv4:    "192.0.2.5",
+		IPv4Len: ipv4PrefixLen,
+		IPv6:    "2001:db8::192:0:2:5",
+		IPv6Len: ipv6PrefixLen,
+	}
+
+	atePort2 = attrs.Attributes{
+		Name:    "atePort2",
+		MAC:     "02:00:01:01:01:02",
+		IPv4:    "192.0.2.6",
+		IPv4Len: ipv4PrefixLen,
+		IPv6:    "2001:db8::192:0:2:6",
+		IPv6Len: ipv6PrefixLen,
+	}
+
+	advertisedIPv41 = ipAddr{address: v41Route, prefix: v4RoutePrefix}
+	advertisedIPv42 = ipAddr{address: v42Route, prefix: v4RoutePrefix}
+	advertisedIPv61 = ipAddr{address: v61Route, prefix: v6RoutePrefix}
+	advertisedIPv62 = ipAddr{address: v62Route, prefix: v6RoutePrefix}
+)
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+type ipAddr struct {
```

Comment:
it seems like you could use the netip.Prefix https://pkg.go.dev/net/netip#Prefix from the standard library instead of defining your own type